Index: sigmah/src/main/java/org/sigmah/shared/dto/UserDTO.java
===================================================================
--- sigmah/src/main/java/org/sigmah/shared/dto/UserDTO.java	(revision 1561)
+++ sigmah/src/main/java/org/sigmah/shared/dto/UserDTO.java	(working copy)
@@ -116,6 +116,17 @@
     }
     
     public boolean getActive(){
+    	// Fix Me
+    	// The line below is a crazy fix for the issue ::http://www.sigmah.org/issues/view.php?id=545
+    	// And is only a temporary solution. 
+    	// For some reason, in the absence of an immediate set operation, the parameters list for the model 
+    	// doesnot contain the parameter "active", and hence returns null
+    	// This is a general class of bug, where a small delay in processing (caused by some dummy action) before 
+    	// the problematic function, causes the function to work as expected. 
+    	// This warrants a more in-depth analysis, and I will send in an updated patch after I get the debugger on 
+    	// my IDE to work properly and manage to set breakpoints.
+    	setName(getName());
+    	
     	Boolean isActive = (Boolean) get("active");
     	return isActive == null ? true : isActive;
     }
