Index: sigmah/src/main/java/org/sigmah/client/page/admin/orgunit/AddOrgUnitWindow.java
===================================================================
--- sigmah/src/main/java/org/sigmah/client/page/admin/orgunit/AddOrgUnitWindow.java	(revision 1561)
+++ sigmah/src/main/java/org/sigmah/client/page/admin/orgunit/AddOrgUnitWindow.java	(working copy)
@@ -97,7 +97,7 @@
         countriesField.setAllowBlank(false);
         countriesField.setValueField("id");
         countriesField.setDisplayField("name");
-        countriesField.setEditable(true);
+        countriesField.setEditable(false);
         countriesField.setEmptyText(I18N.CONSTANTS.flexibleElementDefaultSelectCountry());
         countriesField.setTriggerAction(TriggerAction.ALL);
 
