Index: src/main/java/org/sigmah/shared/dto/element/DefaultFlexibleElementDTO.java
===================================================================
--- src/main/java/org/sigmah/shared/dto/element/DefaultFlexibleElementDTO.java	(revision 1593)
+++ src/main/java/org/sigmah/shared/dto/element/DefaultFlexibleElementDTO.java	(working copy)
@@ -186,7 +186,7 @@
 
 			// Builds the field and sets its value.
 			if (enabled) {
-				final DateField dateField = createDateField(false);
+				final DateField dateField = createDateField(true);
 				dateField.setValue(sd);
 				field = dateField;
 
